### DISCUSSION TITLE
regarding 30 and 50 taxa separate colour legend

### DISCUSSION THREAD

**Discussion by Sam131992** [ChiLiubio/microeco#223] [Q&A]
Dear Microeco Team,
I have forget the script for change the colour of taxa in the relative abundance plot.
I have 30 Phylum and 50 taxa but the colour is limited in the default parameter.
Could you add the script in the message if I have 20, 30 and 50 taxa and I want to give a separate colour on each taxon ow can I do that?
t1$plot_bar(color_values = barplot(1:10, col = color_palette_10), facet = c("Group", "Sample_Name"), xtext_keep = FALSE, legend_text_italic = FALSE, barwidth = 1)
I am trying with this but not working.

**Comment by ChiLiubio** [comment]
Please try to use `randomcoloR` package.
```
library(randomcoloR)
new_color <- randomColor(50)
t1$plot_bar(color_values = new_color, facet = c("Group", "Sample_Name"), xtext_keep = FALSE, legend_text_italic = FALSE, barwidth = 1)
```
It is very hard to have a good visualization effect for so many taxa on the barplot. Why not try to use heatmap.

**Reply by Sam131992** [reply]
Thank you sir for the kind help.

I have two queries regarding environmental correlation analysis:

If we assume we have 30 samples - 15 from Group A and 15 from Group B - and collected 5 soil samples at each field for physicochemical analysis, but for microbiome analysis, we combined these 5 samples into 1 composite sample. So now we have 30 physicochemical observations and 6 microbiome observations. Is it appropriate to analyze the data this way, considering that we have only 6 microbiome observations out of 30 total observations?

How many parameters can we include in the environmental correlation analysis? I have selected 20, but the plot is only showing 8 parameter observations. Can we increase the number of parameters?
